### PR TITLE
Rd 4005 collect new metrics for usage collector

### PR DESCRIPTION
**Pull request by glukhman**
- max_deployments
- max_blueprints
- max_users
- max_tenants
- total _deployments
- total_executions
- total_blueprints
- total_logins
- total_logged_in_users

Where "max..." is the maximum amount of the resource ever to exist at one time in the manager
And "total..." is the total amount of the resource ever to be created in the manager

### CODE REVIEW

File: resources/rest-service/cloudify/migrations/versions/8e8314b1d848_6_2_to_6_3.py [hunk 1]
```diff
@@ -360,3 +364,166 @@ def _add_deployment_resource_tags_column():
...
+    op.execute("""DROP TRIGGER increase_tenants_max ON tenants;""")
+    op.execute("""DROP TRIGGER increase_executions_total ON executions;""")
+    op.execute("""DROP TRIGGER increase_logins_total ON users;""")
+
+    op.execute("""DROP FUNCTION increase_deployments_count();""")
+    op.execute("""DROP FUNCTION increase_blueprints_count();""")
+    op.execute("""DROP FUNCTION increase_users_max();""")
+    op.execute("""DROP FUNCTION increase_tenants_max();""")
+    op.execute("""DROP FUNCTION increase_executions_total();""")
+    op.execute("""DROP FUNCTION increase_logins_total();""")
```
Comment (tehasdf) [inline resources/rest-service/cloudify/migrations/versions/8e8314b1d848_6_2_to_6_3.py:529]: it's a bit ugly that they're all created in a single call but dropped in separate calls

File: resources/rest-service/cloudify/migrations/versions/8e8314b1d848_6_2_to_6_3.py [hunk 2]
```diff
@@ -77,9 +77,13 @@ def upgrade():
     _add_system_properties_column()
     _add_plugins_labels_and_tags_columns()
     _add_deployment_resource_tags_column()
+    _add_usage_collector_columns()
+    _create_usage_collector_triggers()
 
 
 def downgrade():
+    # _drop_usage_collector_triggers()
+    # _drop_usage_collector_columns()
```
Comment (tehasdf) [inline resources/rest-service/cloudify/migrations/versions/8e8314b1d848_6_2_to_6_3.py:86]: zzz

**Comment by tehasdf** [comment]
To run the migrations back and forth easily, for the purposes of testing, you can do something like
```bash
[root@db4ba9e6ff6f /]# cd /opt/manager/resources/cloudify/migrations/
[root@db4ba9e6ff6f migrations]# /opt/manager/env/bin/alembic  current
8e8314b1d848 (head)
[root@db4ba9e6ff6f migrations]# /opt/manager/env/bin/alembic downgrade -1
[root@db4ba9e6ff6f migrations]# /opt/manager/env/bin/alembic current
03ad040e6f78
[root@db4ba9e6ff6f migrations]# /opt/manager/env/bin/alembic upgrade 8e8314b1d848
```

**Comment by tehasdf** [comment]
this quip might be useful for figuring out the starting values:
```python

    uc_table = sa.table('usage_collector',
      sa.column('total_blueprints')
    )
    bps_table = sa.table('blueprints')
    op.execute(
      uc_table.update()
      .values(total_blueprints=
        sa.select([
          sa.func.count(1)
        ])
        .select_from(bps_table)
        .scalar_subquery()
      )
    )
```

File: rest-service/manager_rest/test/endpoints/test_usage_collector_triggers.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
+import os
+from datetime import datetime
+
+from manager_rest.storage import models, user_datastore
+from manager_rest.test import base_test
+from integration_tests.tests.utils import (
+    wait_for_deployment_deletion_to_complete
```
Comment (tehasdf) [inline rest-service/manager_rest/test/endpoints/test_usage_collector_triggers.py:7]: in these tests, executions aren't actually _ran_, so no need to wait for them

Comment (glukhman) [reply rest-service/manager_rest/test/endpoints/test_usage_collector_triggers.py:7]: Yeah, my bad

**Review by tehasdf — DISMISSED** [review]

**Review by geokala — APPROVED** [review]

**Review by tehasdf — DISMISSED** [review]

**Review by tehasdf — APPROVED** [review]

**Pull request merged**